plugins/modules/lvm_pv_move_data.py

```diff
@@ -0,0 +1,221 @@
+            return out.strip()
+
+    def is_pv(device):
+        with pvs_runner("noheadings readonly device", check_rc=False) as ctx:
+            rc, out, err = ctx.run(device=device)
+        return rc == 0
+
+    if not is_pv(source):
+        module.fail_json(msg="Source device %s is not a PV" % source)
+    if not is_pv(destination):
+        module.fail_json(msg="Destination device %s is not a PV" % destination)
+
+    vg_src = run_pvs_command("noheadings vg_name device", source)
+    vg_dest = run_pvs_command("noheadings vg_name device", destination)
+    if vg_src != vg_dest:
+        module.fail_json(
+            msg="Source and destination must be in the same VG. Source VG: '%s', Destination VG: '%s'." % (vg_src, vg_dest)
+        )
+
+    def get_allocated_pe(device):
+        try:
+            return int(run_pvs_command("noheadings pv_pe_alloc_count device", device))
+        except ValueError:
+            module.fail_json(msg="Invalid allocated PE count for device %s" % device)
+
+    allocated = get_allocated_pe(source)
+    if allocated == 0:
+        actions.append('no allocated extents to move')
+    else:
+        # Check destination has enough free space
+        def get_total_pe(device):
+            try:
+                return int(run_pvs_command("noheadings pv_pe_count device", device))
+            except ValueError:
+                module.fail_json(msg="Invalid total PE count for device %s" % device)
+
+        def get_free_pe(device):
+            return get_total_pe(device) - get_allocated_pe(device)
+
+        free_pe_dest = get_free_pe(destination)
+        if free_pe_dest < allocated:
+            module.fail_json(
+                msg="Destination device %s has only %d free physical extents, but source device %s has %d allocated extents. Not enough space." %
+                (destination, free_pe_dest, source, allocated)
+            )
+
+        if module.check_mode:
+            changed = True
+            actions.append('would move data from %s to %s' % (source, destination))
+        else:
+            pvmove_runner = CmdRunner(
+                module,
+                command="pvmove",
+                arg_formats=dict(
+                    auto_answer=cmd_runner_fmt.as_bool("-y"),
+                    atomic=cmd_runner_fmt.as_bool("--atomic"),
+                    autobackup=cmd_runner_fmt.as_fixed("--autobackup", "y" if module.params['autobackup'] else "n"),
+                    verbosity=cmd_runner_fmt.as_func(lambda v: ['-' + 'v' * v] if v > 0 else []),
+                    source=cmd_runner_fmt.as_list(),
+                    destination=cmd_runner_fmt.as_list(),
+                )
+            )
+
+            verbosity = module._verbosity
+            with pvmove_runner("auto_answer atomic autobackup verbosity source destination") as ctx:
+                rc, out, err = ctx.run(
+                    auto_answer=module.params['auto_answer'],
+                    atomic=module.params['atomic'],
```

Comment:
CmdRunner will pick those values directly from module.params, no need to specify them here.
```suggestion
```